lib/events/auditwriter.go

```diff
@@ -533,7 +533,7 @@ func (a *AuditWriter) tryResumeStream() (apievents.Stream, error) {
 			return nil, trace.ConnectionProblem(a.closeCtx.Err(), "operation has been canceled")
 		}
 	}
-	return nil, trace.Wrap(err)
+	return nil, trace.WrapWithMessage(err, "resume attempts exhausted")
```

Comment:
@espadolini TBH I find it a little confusing that we're relying on internal WrapWithMessage's function behavior that it returns non-nil even it `err` is nil (as opposed to regular Wrap). What do you think about making this more explicit e.g. always return an appropriate trace error (we have trace.LimitExceeded that seems fitting) like:
```suggestion
	return nil, trace.LimitExceeded("audit stream resume attempts exhausted: %v", err)
```

Reply:
Thinking about it, it doesn't make sense to mention whatever value is in `err` just because it happened to be the last `err` written to at the end of the loop. Perhaps just this?
```suggestion
	return nil, trace.LimitExceeded("audit stream resume attempts exhausted")
```